src/gluonts/model/wavenet/_estimator.py

```diff
@@ -258,20 +258,22 @@ def __init__(
 
     def train(self, training_data: Dataset) -> Predictor:
         has_negative_data = any(np.any(d["target"] < 0) for d in training_data)
-        mean_length = int(np.mean([len(d["target"]) for d in training_data]))
         low = -10.0 if has_negative_data else 0
         high = 10.0
         bin_centers = np.linspace(low, high, self.num_bins)
         bin_edges = np.concatenate(
             [[-1e20], (bin_centers[1:] + bin_centers[:-1]) / 2.0, [1e20]]
         )
 
-        # Here we override the prediction length for training.
+        # Here we override the prediction length for training if explicitly specified via train_window_length.
         # This computes the loss over longer windows and makes the convolutions more
         # efficient, since calculations are reused.
-        pred_length = min(mean_length, self.train_window_length)
+        pred_length = (
+            self.train_window_length
+            if self.train_window_length is not None
+            else self.prediction_length
+        )
```

Comment:
This definitely mitigates the `InstanceSampler` problem (it's reasonable to have time series of at least length `prediction_length` if you plan on training a model that makes `prediction_length`-long predictions). However, to make this more in line with the `DeepAREstimator` (as well as to take advantage of WaveNet being more efficient at training over longer windows) we could make this `2*self.prediction_length` if not specified otherwise?

Also, `pred_length` is a weird name for this variable here, and this piece of code could really go in the constructor as
```
self.training_window_length = training_window_length if training_window_length is not None else 2*prediction_length
```
so that, from there on, `self.training_window_length` should be used to configure the training network.